### ISSUE TITLE
SSLContext: Cannot assign to class variable "sslsocket_class" via instance

### ISSUE THREAD

**Issue by prauscher** [python/mypy#12308]
**Bug Report**

According to https://docs.python.org/3/library/ssl.html#ssl.SSLContext.sslsocket_class you are encouraged to override `SSLContext.sslsocket_class` to provide your own implementation (e.g. to use fingerprint-checking on certificates). Sadly, mypy detects `SSLContext.sslsocket_class` (probably correctly) as a class variable and setting the variable for a instance results in
```
Cannot assign to class variable "sslsocket_class" via instance
```
However, one might argue that this is a bug in the python standard library. If you could give me hints I will open an issue there.

**To Reproduce**

1. Create file `ssltest.py` 
```python
import ssl


class MyOwnSSLSocket(ssl.SSLSocket):
    pass


context = ssl.create_default_context()
context.sslsocket_class = MyOwnSSLSocket
```
2. Run `mypy ssltest.py`

**Expected Behavior**

```
$ mypy ssltest.py
Success: no issues found in 1 source file
```

**Actual Behavior**

```
$ mypy ssltest.py
ssltest.py:9: error: Cannot assign to class variable "sslsocket_class" via instance
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.920
- Mypy command-line flags: None except the file to test
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.2
- Operating system and version: Linux 5.16.2-arch1-1

**Comment by AlexWaygood** [comment]
Looks like a typeshed bug rather than a mypy or CPython bug, to me. The issue is this line here, where we currently state that `sslsocket_class` is a `ClassVar`. But given the documentation, we probably shouldn't. https://github.com/python/typeshed/blob/master/stdlib/ssl.pyi#L397

**Comment by prauscher** [comment]
Thanks @AlexWaygood for the hint, this issue can than be closed
